src/utils/logger.ts

```diff
@@ -0,0 +1,139 @@
+import { ObserverClass, ObserverIdentifier } from './classObserver';
+import { elapsedTimeInSeconds } from './formatTimeHelpers';
+
+/**
+ * Log configuration
+ */
+ let config = {
+    isEnabled: false,
+    logWithStackTrace: false,
+    notifyLogSubscribers: false,
+    isDevelopment: false,
+    isMSALLoggingEnabled: false
+} as LoggerConfiguration;
+
+/**
+ * Internal helper class to keep track of any log-subscribers.
+ */
+class LogSubscribers extends ObserverClass {
+    constructor() {
+        super();
+    }
+}
+// The ObserverClass needs a "type" on each subscription. Consumers of the logger (i.e. logSubscriptions) do not 
+// need to know about this and we just assign the same "type" to all subscribers; this one. 
+const LogSubscriberType : string  = 'LogSubscriberType';
+const logSubscribers : LogSubscribers = new LogSubscribers();
+
+/**
+ * Add a subscriber to the given type. If the configuration is setup to notify subscribers, then the callback
+ * given here will be called if a logging is performed on the given type.
+ * @param callback 
+ * @returns a unique identifier for the subscriber which later can be used in {@link removeLogSubscriber}.
+ */
+export function addLogSubscriber(callback: Function) : ObserverIdentifier {
```

Comment:
Vi bør definere hvordan denne callback funksjonen ser ut. Denne bør ta inn message string/args, og også LogType. Noen Subscribers vil nok vite logType og reagere annerledes på info, error og warning.

Reply:
👍 
Men hvis OK så tror jeg neste iterasjon kan gjøres i en separat issue/PR, så får jeg avsluttet alt for nåværende sprint?

Reply:
Function er det samme som any, og bør fikses. Utenforståendende vet ikke hvordan denne funksjonen skal se ut nå.

@Noggling Skal vi ha egen PR på å legge til logType, eller vil du ha egen US på dette? Tror ikke det er så veldig mye jobb å få lagt til.

Reply:
Hvis det ikke er mye arbeid ser jeg helst at det legges til i denne RP, men hvis det blir for mye så lager jeg ny US. men da vil den funksjonaliteten ikke bli tilgjengelig med det første. Men enig med @OveBDev Function må vekk ;)

det kan være en:

```
type LogTypeMessage = (key: string, message: string) => void
type LogTypeError = (key: string, error: BaseError) => void

type LogType = LogTypeMessage | LogTypeError osv....
```

Reply:
Har i hvert fall fått en eksplisitt interface på callback-funksjonen nå - men ser at å skrive om subscription på spesifikke log-typer trenger litt mer jobb og kan for min del gjerne trekkes ut til en egen issue (har man allerede konkrete bruksområder, eller er dette bare nice-to-have-i-tilfelle-noen-trenger-det? :) )